### PR TITLE
Move blocks feature

### PR DESCRIPTION
**Pull request by Horqu**

### CODE REVIEW

File: controllers/game_controller.py [hunk 1]
```diff
@@ -9,9 +9,9 @@ def __init__(self, previousView):
     def save_result(self):
         pass
 
+    # PLACEHOLDER METHOD TO SHOW ALL BLOCKS
```
Comment (Szynkaa) [inline controllers/game_controller.py:12]: duplication of comment

File: interactors/rotate_block_interactor.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from entities.gameboard_entity import Gameboard
+from entities.block_entity import Block
+
+class RotateBlockInteractor():
+    def __init__(self, gameboard: Gameboard):
+        self.gameboard = gameboard
+
+    def rotate_block(self, block: Block):
+        print('rotate')
```
Comment (ptakpiotr) [inline interactors/rotate_block_interactor.py:9]: This print statement will be replaced later on with appropriate logic?

File: entities/gameboard_entity.py [hunk 3]
```diff
@@ -5,4 +5,9 @@ def __init__(self, master, width, height):
         super().__init__(master, width=width, height=height, background='gray75')
 
         self.width = width
-        self.height = height
\ No newline at end of file
+        self.height = height
+        self.block_width = 25
+        self.game_matrix_width = width // self.block_width
+        self.game_matrix_height = height // self.block_width
+        self.game_matrix = [[(0, '', 'na')] * self.game_matrix_width for _ in range(self.game_matrix_height)] # non-active(na) or active(a)
```
Comment (Szynkaa) [inline entities/gameboard_entity.py:12]: maybe boolean instead of 'na'

File: interactors/drop_block_interactor.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+from entities.gameboard_entity import Gameboard
+from entities.block_entity import Block
+
+class DropBlockInteractor():
+    def __init__(self, gameboard: Gameboard):
+        self.gameboard = gameboard
+
+    def drop_block(self, block: Block):
+
```
Comment (Szynkaa) [inline interactors/drop_block_interactor.py:11]: pls remove this empty line

File: interactors/drop_block_interactor.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        can_move = True
+        while can_move:
+
+            lowest_active_blocks = []
+            for j in range(self.gameboard.game_matrix_width):
+                for i in reversed(range(self.gameboard.game_matrix_height)):
+                    if self.gameboard.game_matrix[i][j][2] == 'a':
+                        lowest_active_blocks.append((i, j))
+                        break  # Break the inner loop when you find an active block 
+            can_move = all(i + 1 < self.gameboard.game_matrix_height and self.gameboard.game_matrix[i + 1][j][0] == 0
```
Comment (Szynkaa) [inline interactors/drop_block_interactor.py:22]: but add some empty line here :D (to separate for from next instruction)

File: interactors/rotate_block_interactor.py (hunk 2 above)
Comment (Horqu) [reply interactors/rotate_block_interactor.py:9]: Yup, it is only a placeholder

File: interactors/drop_block_interactor.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+                for j in range(self.gameboard.game_matrix_width):
+                    for i in reversed(range(self.gameboard.game_matrix_height)):  # We move from bottom to top
+                        if self.gameboard.game_matrix[i][j][2] == 'a' and i + 1 < self.gameboard.game_matrix_height:  # We add a condition to prevent exceeding the lower boundary
+                            value, block_type, status = self.gameboard.game_matrix[i][j]
+                            # Move the block down
+                            self.gameboard.game_matrix[i + 1][j] = self.gameboard.game_matrix[i][j]
+                            self.gameboard.game_matrix[i][j] = (0, '', 'na')  # Remove the old block
+
+        # Set the lowest active blocks to inactive
+        for i, j in active_blocks:
```
Comment (Szynkaa) [inline interactors/drop_block_interactor.py:39]: comment and code inconsistent: lowest active blocks vs active blocks

File: interactors/move_block_interactor.py [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+            for j in range(self.gameboard.game_matrix_width):
+                for i in reversed(range(self.gameboard.game_matrix_height)):  # We move from bottom to top
+                    if self.gameboard.game_matrix[i][j][2] == 'a' and i + 1 < self.gameboard.game_matrix_height:  # We add a condition to prevent exceeding the lower boundary
+                        value, block_type, status = self.gameboard.game_matrix[i][j]
+                        # Move the block down
+                        self.gameboard.game_matrix[i + 1][j] = self.gameboard.game_matrix[i][j]
+                        self.gameboard.game_matrix[i][j] = (0, '', 'na')  # Remove the old block
+        else:
+            # Set the lowest active blocks to inactive
+            for i, j in active_blocks:
```
Comment (Szynkaa) [inline interactors/move_block_interactor.py:35]: inconsistent

File: controllers/game_controller.py (hunk 1 above)
Comment (Horqu) [reply controllers/game_controller.py:12]: Deleted

File: entities/gameboard_entity.py (hunk 3 above)
Comment (Horqu) [reply entities/gameboard_entity.py]: Changed values to [Is the block filled?, Block color, Is the block active? (currently moving)]

File: interactors/drop_block_interactor.py (hunk 4 above)
Comment (Horqu) [reply interactors/drop_block_interactor.py]: done

File: interactors/drop_block_interactor.py (hunk 6 above)
Comment (Horqu) [reply interactors/drop_block_interactor.py:38]: done

File: interactors/drop_block_interactor.py (hunk 5 above)
Comment (Horqu) [reply interactors/drop_block_interactor.py]: done

File: interactors/move_block_interactor.py (hunk 7 above)
Comment (Horqu) [reply interactors/move_block_interactor.py:35]: done

**Review by Szynkaa — APPROVED** [review]

**Review by ptakpiotr — APPROVED** [review]

**Pull request merged**